### DISCUSSION TITLE
MPC and Logic Pro USB MIDI Assistance

### DISCUSSION THREAD

**Discussion by AndyLAndrew** [probonopd/MiniDexed#270] [Q&A] [answered]
Hello Everyone! First wanted to say thank you to probonopd and all others who contributed/contributing to this! Absolutely amazing project and I can't wait to see what else is in store!
----
On to the question: so I have a set-up involving a Moog Matriarch acting as a midi keyboard for my MPC One, with the One processing all input for any other midi devices connected. I'm able to control Minidexed over USB using the Matriarch, assuming that means it is Class Compliant, but I cannot get minidexed recognized by the MPC. Does this mean the MPC itself is not class compliant, and I will have to install a MIDI-DIN port (the MPC has standard MIDI In/Out)?

I'm hoping to not have to rely on DIN as I just migrated my setup to USB only haha, but understand if that might be the case.

---
As for Logic, it's a similar question but this one I'm certain requires a MIDI interface since mac requires usb drivers correct?

---
Thank you and apologies if the questions are easy fixes! Never thought a USB-MIDI setup would bring more issues for myself than a classic DIN one!

**Comment by luisfcorreia** [comment, marked as answer]
Akai MPC ONE is probably NOT class compliant

(as any device requiring drivers)

**Reply by Banana71** [reply]
My favorite solution would be a Raspberry Pico with a MIDI library.
The Pico should actually be recognized by every host. (Class Compliant)
Does anyone already have experience with this?
It is possible to connect directly the UART-Pins? From Pico RX and TX to Raspberry Pi TX and RX?
https://diyelectromusic.wordpress.com/2021/01/23/midi-micropython-and-the-raspberry-pi-pico/

**Reply by diyelectromusic** [reply]
It most definitely is possible :)

Here I'm using a Pico to select the voice for the MiniDexed by connecting RX/TX on the Pico and the Pi.  It also has a built-in MIDI serial to USB too: https://diyelectromusic.wordpress.com/2022/05/20/pi-pico-midi-button-voice-select/

So the Pico will act as a USB MIDI device routing to/from the MiniDexed which requires a USB MIDI host. I'd hope the Pico would just be seen as a USB MIDI device to the MPC ONE...?

If you just want the USB to serial MIDI, then here is a similar setup using a Xaio, but as it's CircuitPython should be trivially converted over to a Pico too: https://diyelectromusic.wordpress.com/2022/04/21/circuitpython-usb-to-serial-midi-router/

This is all in CircuitPython though, so I'm still making up my mind about the performance... an alternative solution could involve an Arduino with a ATmega32U4, but the 16Mhz one (which is ideal for MIDI routing) is a 5V device, so you can't connect directly to the Pi, but the 3V3 device is only 8MHz, so again I don't know if that would have performance issues...

For completeness, here is a C/Circle implementation of USB MIDI to serial MIDI on a Pico too: https://github.com/rsta2/pico/tree/main/midi_adapter

Kevin

**Reply by AndyLAndrew** [reply]
awesome, thanks everyone! Bummer the MPC Standalone isn't class compliant for USB output, but luckily it does have MIDI DIN so I can just add a classic MIDI out following the instructions under the hardware tab on the wiki. I do like the Pico implementation since I'm assuming it could be mounted inside the case and provide a pseudo MIDI USB IN right? Also provided the Pico can provide a USB driver, yes it should be recognized by the MPC. It's pretty seamless with all other USB MIDI hardware and USB audio interfaces so I was a bit surprised when I figured out it couldn't work with Minidexed.

Definitely will keep this in mind if I ever want to take Minidexed in my pocket haha. I have friends asking me to build them little boxes so maybe that will be a better option than DIN (they aren't very keen on supporting classic MIDI either).
